src/oedisi/componentframework/system_configuration.py

```diff
@@ -152,17 +155,21 @@ def clean_model(self, target_directory="."):
             if proc.name() == "helics_broker":
                 proc.kill()
 
-    @validator("components")
+    @field_validator("components")
+    @classmethod
     def check_component_names(cls, components):
         "Check that the components all have unique names"
         names = set(map(lambda c: c.name, components))
         assert len(names) == len(components)
         return components
 
-    @validator("links")
-    def check_link_names(cls, links, values, **kwargs):
-        if "components" in values:
-            components = values["components"]
+    # TODO[pydantic]: We couldn't refactor the `validator`, please replace it by `field_validator` manually.
```

Comment:
Dumb question: hasn't this replacement already been performed?

Reply:
Should be removed now. That TODO was autogenerated and I replaced it myself.